src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -843,20 +903,27 @@ private static List<CompletionResult> GetParameterCompletionResults(
                     {
                         foreach (var attr in compiledAttributes)
                         {
-                            var pattr = attr as ParameterAttribute;
-                            if (pattr != null && pattr.DontShow)
+                            if (attr is not ParameterAttribute pattr)
+                            {
+                                continue;
+                            }
+
+                            helpMessage = GetParameterHelpMessage(pattr, commandAssembly);
+                            if (pattr.DontShow)
```

Comment:
If DontDhow why do we assign `helpMessage`?
And maybe store the attribute and call GetParameterHelpMessage exactly where we use the helpMessage?
And maybe do all normalizations of result in GetParameterHelpMessage? I see `string.IsNullOrWhiteSpace` below. We could always return null if message is "blank".

Reply:
Sorry I updated the commit as I wasn't doing it correctly. If `DontShow` we don't emit anything and the `GetParameterHelpMessage` is done after now.